### PR TITLE
conversion of odt to vtt file

### PR DESCRIPTION
**Pull request by AlokSP**

### CODE REVIEW

File: src/main/java/org/spoken_tutorial/health/elasticsearch/JsonService/JsonService.java [hunk 1]
```diff
@@ -345,4 +346,46 @@ public String convertHtmltoOdt(String htmlFie, int tutorialId) {
...
+                Path vttDir = Paths.get(mediaRoot, Config.uploadDirectoryTimeScriptvttFile);
+
+                try {
+
+                    Files.createDirectories(vttDir);
+                    Path vttPath = Paths.get(mediaRoot, Config.uploadDirectoryTimeScriptvttFile, tutorialId + ".vtt");
+                    logger.info("Converting odt file to vtt file.... scriptPath:{}, vttPath:{}", odtFilePath.toString(),
+                            vttPath.toString());
+                    String extractedText;
+                    extractedText = ServiceUtility.extractTextFromFile(odtFilePath);
```
Comment (sunilshetye) [inline src/main/java/org/spoken_tutorial/health/elasticsearch/JsonService/JsonService.java:378]:
```suggestion
                    String extractedText = ServiceUtility.extractTextFromFile(odtFilePath);
```
Combine the two.

File: src/main/java/org/spoken_tutorial/health/elasticsearch/config/ServiceUtility.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                    continue;
+                }
+                if (line.matches("^([01]?\\d|2[0-3]):([0-5]?\\d)$")
+                        || line.matches("^([01]?\\d|2[0-3])\\.([0-5]?\\d)$")) {
+                    textFlag = false;
+
+                    String newtimeValue = "00:" + formatToTimeScript(line);
+
+                    if (!startTimeString.equals("00:")) {
+                        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm:ss");
```
Comment (sunilshetye) [inline src/main/java/org/spoken_tutorial/health/elasticsearch/config/ServiceUtility.java:79]: Add this formatter at the beginning of this function. It should be declared before the for loop.

File: src/main/java/org/spoken_tutorial/health/elasticsearch/config/ServiceUtility.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                        String endTimeString = endTime.toString();
+                        sb.append(startTimeString + " --> " + endTimeString + "\n");
+                        sb.append(textValue + "\n\n");
+                        index++;
+                    }
+                    startTimeString = newtimeValue + ".000";
+
+                } else {
+                    if (textFlag) {
+                        textValue = textValue + line;
```
Comment (sunilshetye) [inline src/main/java/org/spoken_tutorial/health/elasticsearch/config/ServiceUtility.java:93]:
```suggestion
                        textValue = textValue + " " + line;
```
Are you combining lines here? if yes, put a space between the lines. Trim if required.

**Review by sunilshetye — CHANGES_REQUESTED** [review]
Please check the comments,

File: src/main/java/org/spoken_tutorial/health/elasticsearch/config/ServiceUtility.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+                    if (!startTimeString.equals("00:")) {
+
+                        LocalTime endTime = LocalTime.parse(newtimeValue, formatter);
+
+                        endTime = endTime.minusNanos(500_000_000);
+
+                        String endTimeString = endTime.toString();
+                        sb.append(startTimeString + " --> " + endTimeString + "\n");
+                        sb.append(textValue + "\n\n");
```
Comment (sunilshetye) [inline src/main/java/org/spoken_tutorial/health/elasticsearch/config/ServiceUtility.java:87]: After appending textValue to sb, should you reset textValue to empty string and textFlag to false? Please check.

**Review by sunilshetye — CHANGES_REQUESTED** [review]
Check if textValue and textFlag should be reset after appending to sb.

File: src/main/java/org/spoken_tutorial/health/elasticsearch/config/ServiceUtility.java (hunk 4 above)
Comment (AlokSP) [reply src/main/java/org/spoken_tutorial/health/elasticsearch/config/ServiceUtility.java:87]: In else part we are reseting textValue by assigning line.trim() and setting the textFlag true. 
textValue = line.trim();
textFlag = true;

Comment (sunilshetye) [reply src/main/java/org/spoken_tutorial/health/elasticsearch/config/ServiceUtility.java:87]: you have code to set it to true. do you not need to reset it to false when the next timestamp starts?

Comment (AlokSP) [reply src/main/java/org/spoken_tutorial/health/elasticsearch/config/ServiceUtility.java:87]: it is setting false too, when line contains timestamp
if (line.matches("^([01]?\\d|2[0-3]):([0-5]?\\d)$")
                        || line.matches("^([01]?\\d|2[0-3])\\.([0-5]?\\d)$")) {
                    textFlag = false;

Comment (sunilshetye) [reply src/main/java/org/spoken_tutorial/health/elasticsearch/config/ServiceUtility.java:87]: okay, thanks

**Review by sunilshetye — APPROVED** [review]

**Pull request merged**